### PR TITLE
[WBCAMS-220] gracefully handle BadHttpRequestException rev 2

### PR DESCRIPTION
**Pull request by jonathan-longe**
Hold until changes associated with the removal of Jobs without salary is complete.   This changes shown in this PR request have been previously reviewed and approved but were reverted from master to minimize the changes associated with a production release.  This PR has been created so the changes can be merged into master when appropriate.

### CODE REVIEW

**Comment by jonathan-longe** [comment]
This PR was previously approved.  Waiting for client go-ahead before merging.

**Review by infojunkie — APPROVED** [review]
Lgtm, thanks

**Comment by infojunkie** [comment]
Is this still needed or can we close this PR?

**Pull request closed without merging**
